src/libcore/num/mod.rs

```diff
@@ -329,6 +329,27 @@ $EndFeature, "
             }
         }
 
+        doc_comment! {
+            concat!("Returns the number of leading ones in the binary representation of `self`.
+
+# Examples
+
+Basic usage:
+
+```
+", $Feature, "let n = 0b10110011", stringify!($SelfT), ";
+
+assert_eq!(n.leading_ones(), 1);",
```

Comment:
I don't think this will pass -- this will have no leading ones for any type size but `u8`.